app/controllers/rooms_controller.rb

```diff
@@ -51,6 +52,7 @@ def index
   # GET /rooms/1
   # GET /rooms/1.json
   def show
+    redirect_to rooms_path unless @room.visible || session[:user_admin]
```

Comment:
Should this have an additional condition of @room.building.visible to prevent non admins form directly going to a room if the building is NOT visible?